solr/core/src/java/org/apache/solr/response/transform/GeoTransformerFactory.java

```diff
@@ -144,88 +148,111 @@ public void transform(SolrDocument doc, int docid) throws IOException {
 
     }
     
+    // if source has been renamed, update reference
```

Comment:
Yes, basically the fact that this class had been doing renaming _outside the context of "simple" renaming that corrected for interactions with other field renames_ was a latent problem (perhaps I could/should add a test to stress this issue -- though I might leave that for another issue?). I'm going to think about this, and the "`FieldRenamer` interface vs. "subclass of `TransformerFactory`" question -- perhaps there's a way to factor some of this logic out so it can live in once place and not be the responsibility of each `FieldRenamer` implementation. I'll report back.

Reply:
(I just pushed commits addressing all other feedback but this subclass/interface question -- oh, and did also add tests to stress the renaming issue, as I suggested immediately above)

wrt the subclass/interface question: on further reflection, unfortunately I think it might just make things more convoluted (and less flexible) to try to factor out the "renaming" logic, as opposed to (as currently implemented) delegating that responsibility to each class that implements `FieldRenamer` (at a minimum each DocTransformer needs to parse its own SolrParams to determine "source" field, and the GeoTransformer in some cases _doesn't_ rename a field (instead retrieving source values directly from docValues) -- and once you add hooks for both those cases, it's unclear that anything is really being simplified). 

Unless you feel strongly about it, I'm also inclined to leave this as an interface, because I don't think there's any direct benefit (in terms of factoring out code) to making it a subclass, and although the subclass approach would clarify somewhat the main use of this type, it would also needlessly prevent users from extending existing `TransformerFactory` classes to add support for field renaming. This might be kind of an edge case, but absent a strong preference for strict subclass, I'll probably stick with the flexibility of an interface-based approach.